presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java

```diff
@@ -99,7 +101,11 @@ public ByteCodeNode generateExpression(Signature signature, ByteCodeGeneratorCon
         Scope scope = generatorContext.getScope();
 
         ByteCodeNode switchBlock;
-        if (constantValues.size() < 1000) {
+
+        if (maxValue <= Integer.MAX_VALUE && minValue >= Integer.MIN_VALUE) {
+            switchBlock = new Block().comment("value between max and min integer").dup(javaType).append(ByteCodeUtils.boxPrimitive(javaType)).longToInt();
```

Comment:
This seems to be missing the actual `switch` code. Can you make sure this case is covered by the existing tests and if not, add a test?